guide.js

```diff
@@ -264,15 +286,17 @@ function userInputData(rowHtml, checkboxElement){
 	recalculateProgressAndUpdateProgressUI();
 }
 
-
+//These variables are used to make sure we don't run a ton of refresh code constantly.
 var __displayingIframe = null;
+var __linkFormatType = null; //if for example the iframe was already off and we now have it set to window, the window link rewrite code wouldn't run without checking this.
```

Comment:
why do we need both?
User can't change this setting on the guide page, so they will have to reload the page anyways for it to take effect.

Reply:
I'm thinking of restructuring this entire function anyway, but the main reason I left both there for now was options in case we ever wanted to be able to change the guide options directly on the guide page like we have with the map.

The restructuring I had in mind is to have the show/hide and the link target rewriting be two separate things, so the displayingIframe variable would be it's own if/else conditional followed by the linkFormatType if/else rather than escaping the function at the start. Seems easier for programmers to figure out what the function is doing that way since the logic goes more linearly to a "did this setting change? what value is this setting now? change these things based on that value"

Reply:
>have the show/hide and the link target rewriting be two separate things
 They are inextricably linked though.
 if we hide the iframe, we no longer want things to open in the iframe.
 > options in case we ever wanted to be able to change the guide options directly on the guide page like we have with the map
 Nope. We will switch to a framework/backend reorganization before we do this. Some of the backend rewrite will allow us to do these changed settings easier, so we can deal with it then when we are already rewriting it.

Reply:
> They are inextricably linked though. if we hide the iframe, we no longer want things to open in the iframe.

Oh my bad I forgot to explain the actual primary reason I added this variable in lol. Link rewrite and frame toggling are not technically the same thing anymore. The one situation this new variable accounts for is if the user already has the iframe setting to be "off", which sets all iframe links to "_blank" and turns the frame off, then decides to switch to "Other Window" which turns the frame off and sets all iframe links to "externalSecondWindow". Without this extra variable, since the iframe is already off prior to changing to "Other Window" it would exit the function without updating the links. I had to change this because the new UI system only hides the iframe when it is toggled off instead of completely destroying and rebuilding it like it used to. This means the "myframe" target can't be used situationally now as the target for both settings since the "myframe" target still links to the hidden iframe even in "Other Window" mode. That's why I was thinking of having them be separate functions in the future, because the new setup actually does require checking their logic separately anyway. The support for the ability to change settings on the actual page is sort of a bonus, idk why I didn't explain this other part to begin with.